sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -126,6 +142,30 @@ public boolean isEnabled() {
     return sentryId;
   }
 
+  @Override
+  public void captureEnvelope(SentryEnvelope envelope, @Nullable Object hint) {
+    if (!isEnabled()) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "Instance is disabled and this 'captureEnvelope' call is a no-op.");
+    } else if (envelope == null) {
+      options.getLogger().log(SentryLevel.WARNING, "captureEnvelope called with null parameter.");
+    } else {
```

Comment:
for this case, it would be ok because it's not a public API, but all the public API should not throw exception or clients will need to wrap up on try/catch for safety, will do it.